File: backend/src/AssetBridge.Domain/Entities/Representatives/Representative.cs
using AssetBridge.Domain.Entities.Common;
using AssetBridge.Domain.Entities.Users;
using AssetBridge.Domain.Enums;

namespace AssetBridge.Domain.Entities.Representatives;

// Represents a vetted local on-the-ground coordinator in Sri Lanka acting on behalf of overseas owners.
// Connects to User identity without duplicating authentication credentials.
public class Representative : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string FullName { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string District { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string? Address { get; set; }
    public string? NationalIdNumber { get; set; }
    public VerificationStatus VerificationStatus { get; set; } = VerificationStatus.Pending;
    public string? VerificationNotes { get; set; }
    public string? Bio { get; set; }
    public bool IsActive { get; set; } = true;
}

File: frontend-web/src/types/representative.ts
export type VerificationStatus = 'Pending' | 'Verified' | 'Rejected' | 'Expired';

export interface RepresentativeResponseDto {
  id: string;
  userId: string;
  fullName: string;
  phoneNumber: string;
  email: string;
  district: string;
  city: string;
  address?: string;
  nationalIdNumber?: string;
  verificationStatus: VerificationStatus;
  verificationStatusName: string;
  verificationNotes?: string;
  bio?: string;
  isActive: boolean;
  createdAtUtc: string;
  updatedAtUtc?: string;
}

export interface CreateRepresentativeRequestDto {
  fullName: string;
  phoneNumber: string;
  email: string;
  district: string;
  city: string;
  address?: string;
  nationalIdNumber?: string;
  bio?: string;
}

export interface UpdateRepresentativeRequestDto {
  fullName: string;
  phoneNumber: string;
  district: string;
  city: string;
  address?: string;
  nationalIdNumber?: string;
  bio?: string;
  isActive?: boolean;
}

export interface UpdateVerificationRequestDto {
  status: VerificationStatus;
  notes?: string;
}

export interface RepresentativeQueryParametersDto {
  pageNumber?: number;
  pageSize?: number;
  searchTerm?: string;
  district?: string;
  city?: string;
  status?: VerificationStatus | string;
  isActive?: boolean;
}
